docs/self-managed/setup/deploy/amazon/amazon-eks/terraform-setup.md

```diff
@@ -250,65 +227,419 @@ access_entries = {
+
+:::
+
+#### Step 1: create a configuration file for OpenSearch
+
+1. In the folder where your `config.tf` file resides, create a new file named `opensearch.tf`.
+2. Add the following content to `opensearch.tf` to make use of the provided OpenSearch module:
+
+<Tabs groupId="opensearch-tf">
+  <TabItem value="standard" label="Standard" default>
+
+:::caution Network based security
+
+The standard deployment for OpenSearch relies on the first layer of security, which is the Network.
+
+While this setup allows easy access, it may expose sensitive data. To enhance security, consider implementing IAM Roles for Service Accounts (IRSA) to restrict access to the OpenSearch cluster, providing a more secure environment.
+
+For more information, see the [Amazon OpenSearch Service Fine-Grained Access Control documentation](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/fgac.html#fgac-access-policies).
+:::
+
+```hcl reference
+https://github.com/camunda/camunda-tf-eks-module/blob/feature/opensearch-doc/examples/camunda-8.6/opensearch.tf
 ```
 
-3. Run `terraform apply` again to print the outputs in the terraform state.
+  </TabItem>
+  
+  <TabItem value="irsa" label="IRSA">
+
+In addition to traditional username and password authentication, you can also use [**IRSA (IAM Roles for Service Accounts)**](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) to securely connect to OpenSearch. IRSA enables your Kubernetes workloads to assume IAM roles without managing AWS credentials directly.
+
+**Note**: Using IRSA is optional. If you prefer, you can continue using password-based access to your OpenSearch domain.
 
-We can now export those values to environment variables to be used by Helm charts:
+If you choose to use IRSA, you’ll need to take note of the **IAM role name** created for OpenSearch and the **AWS Account ID**, as these will be required later to annotate the Kubernetes service account.
+
+##### OpenSearch IRSA role and policy
+
+To configure IRSA for OpenSearch, the OpenSearch module uses outputs from the EKS cluster module to define the necessary IAM role and policies.
+
+Here's an example of how to define the IAM role trust policy and access policy for OpenSearch:
+
+```hcl reference
+https://github.com/camunda/camunda-tf-eks-module/blob/feature/opensearch-doc/examples/camunda-8.6-irsa/opensearch.tf
+```
+
+Once the IRSA configuration is complete, ensure you **record the IAM role name** (from the `iam_opensearch_role_name` configuration) and the **AWS Account ID** (from `module.eks_cluster.aws_caller_identity_account_id`). You will need these to annotate the Kubernetes service account in the next step.
+
+This configuration will deploy an OpenSearch domain with advanced security enabled. You must provide your own username (`advanced_security_master_user_name`) and password for the master user (`advanced_security_master_user_password`).
```

Comment:
do I even need username / password?
It feels counterintuitive to talk about IRSA being more secure, but then we slap username/password on top.